<launch>
    <!-- Arguments -->
    <arg name="use_case_yaml" default="arenosillo.yaml"/>
    <arg name="rviz_name"  default="arenosillo"/>
    <rosparam command="load" file="$(find usp_testers)/config/$(arg use_case_yaml)" />

    <!-- RVIZ -->
    <node pkg="rviz" type="rviz" name="rviz_node" args="-d $(find usp_testers)/config/$(arg rviz_name).rviz" required="true"/>
    <!-- GAUSS -->
    <node pkg="db_manager" type="DBmanager" name="data_base_manager" output="screen" required="true"/>
    <node pkg="emergency_management" type="emergency_management.py" name="emergency_management" required="true"/>
    <node pkg="tracking" type="tracking" name="tracking" required="true"/>
    <node pkg="db_manager" type="visualizer.py" name="visualizer" output="screen" required="true"/>
    <node pkg="usp_manager" type="USPmanager" name="usp_manager" output="screen" required="true"/>
    <node pkg="rqt_gui" type="rqt_gui" name="rqt_gauss" args="--perspective-file $(find rqt_gauss)/config/gauss.perspective"/>
    <node pkg="gauss_light_sim" type="gauss_light_sim_node" name="gauss_light_sim_node" output="screen" required="true">
        <param name="timing_file" value="$(find usp_testers)/config/$(arg use_case_yaml)"/>
    </node>
    <node pkg="tactical_deconfliction" type="ConflictSolver" name="tactical_deconfliction" output="screen" required="true"/>
    <node pkg="monitoring" type="monitoring" name="monitoring" required="true" output="screen"/>

</launch>
